client/beefy/src/gossip.rs

```diff
@@ -52,13 +49,52 @@ where
 /// A type that represents hash of the message.
 pub type MessageHash = [u8; 8];
 
-type KnownVotes<B> = BTreeMap<NumberFor<B>, fnv::FnvHashSet<MessageHash>>;
+struct KnownVotes<B: Block> {
+	last_done: Option<NumberFor<B>>,
+	live: BTreeMap<NumberFor<B>, fnv::FnvHashSet<MessageHash>>,
+}
+
+impl<B: Block> KnownVotes<B> {
+	pub fn new() -> Self {
+		Self { last_done: None, live: BTreeMap::new() }
+	}
+
+	/// Create new round votes set if not already present.
+	pub fn insert(&mut self, round: NumberFor<B>) {
+		if !self.live.contains_key(&round) {
```

Comment:
I'd replace with `self.live.entry(number).or_default();` :) But this version is fine too :)